### PR TITLE
Check specific app installation path in app_auth_roundtripper

### PR DESCRIPTION
**Pull request by smg247**
Since https://github.com/kubernetes/test-infra/pull/27181 was merged adding the JWT to the `/repos/org/repo/installation` path we have noticed an error when accessing the `getRepo` endpoint with a repo named `integr8ly/installation`. It seems that simply checking for the `installation` suffix will not suffice since that can be a repo name. If we use a specific regex instead we won't hit false positives.

### CODE REVIEW

File: prow/github/app_auth_roundtripper_test.go [hunk 1]
```diff
@@ -372,6 +372,32 @@ func TestAppsAuth(t *testing.T) {
...
+			},
+			responses: map[string]*http.Response{"/repos/kuber/k8s/installation": {
+				StatusCode: 200,
+				Body:       serializeOrDie(AppInstallation{}),
+			}},
+			verifyRequests: func(r []*http.Request) error {
+				if n := len(r); n != 1 {
+					return fmt.Errorf("expected exactly one request, got %d", n)
+				}
+				if val := r[0].Header.Get("Authorization"); !strings.HasPrefix(val, "Bearer ") || val == "Bearer the-token" {
```
Comment (alvaroaleman) [inline prow/github/app_auth_roundtripper_test.go:392]: This test currently verifies that your change does not work, `the-token` is an access token to the `kuber` installation, not a JWT :) The reason for this is that the regex doesn't match correctly.

Comment (smg247) [reply prow/github/app_auth_roundtripper_test.go:392]: The condition is a little confusing looking, but it's actually checking that the token is **not** "the-token". I can split it into separate conditions if that makes it clearer. I actually verified the correctness here both in the debugger and by pulling these changes into my downstream tool and testing the functionality.

Comment (alvaroaleman) [reply prow/github/app_auth_roundtripper_test.go:392]: Okay, then I am confused as to how this works as the regexp does not match `/repos/kuber/k8s/installation`: https://regex101.com/r/onRTB2/1 ..?

Comment (smg247) [reply prow/github/app_auth_roundtripper_test.go:392]: The first `\` in each `\\w*` section is just an escape. Try https://regex101.com/r/dUEeQd/1

Comment (smg247) [reply prow/github/app_auth_roundtripper_test.go:392]: I just updated it to use back ticks instead to make it more legible.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**